misk/src/main/kotlin/misk/client/TypedHttpClientModule.kt

```diff
@@ -163,7 +155,49 @@ private abstract class TypedClientFactory<T : Any>(
   @Inject(optional = true)
   private val eventListenerFactory: EventListener.Factory? = null
 
-  fun typedClient(client: OkHttpClient, baseUrl: String): T {
+  @Inject private lateinit var httpClientConfigUrlProvider: HttpClientConfigUrlProvider
+
+  @Inject private lateinit var httpClientFactory: HttpClientFactory
+
+  /**
+   * Build up a typed client dynamically in runtime. This is useful for platform-type services
+   * that cannot statically define all of the services they talk to.
+   *
+   * Services should cache the resulting clients to avoid incurring the construction on every call.
```

Comment:
misk should probably provide this, just not at this level.

Reply:
yeah, maybe. I'll build it for plasma and see